code/datums/elements/turf_transparency.dm

```diff
@@ -1,29 +1,33 @@
 
 /datum/element/turf_z_transparency
 	element_flags = ELEMENT_DETACH
-	var/show_bottom_level = FALSE
-	var/is_openspace = FALSE
 
 ///This proc sets up the signals to handle updating viscontents when turfs above/below update. Handle plane and layer here too so that they don't cover other obs/turfs in Dream Maker
-/datum/element/turf_z_transparency/Attach(datum/target, show_bottom_level = TRUE, is_openspace = FALSE)
+/datum/element/turf_z_transparency/Attach(datum/target)
 	. = ..()
 	if(!isturf(target))
 		return ELEMENT_INCOMPATIBLE
 
 	var/turf/our_turf = target
 
-	src.show_bottom_level = show_bottom_level
-
-	if(is_openspace)
+	our_turf.layer = OPENSPACE_LAYER
+	if(istype(our_turf, /turf/open/openspace))
 		our_turf.plane = OPENSPACE_PLANE
-		our_turf.layer = OPENSPACE_LAYER
 
 	RegisterSignal(target, COMSIG_TURF_MULTIZ_DEL, .proc/on_multiz_turf_del)
 	RegisterSignal(target, COMSIG_TURF_MULTIZ_NEW, .proc/on_multiz_turf_new)
 
 	ADD_TRAIT(our_turf, TURF_Z_TRANSPARENT_TRAIT, ELEMENT_TRAIT(type))
 
-	update_multiz(our_turf, TRUE, TRUE)
+	var/turf/below_turf = our_turf.below()
+	if(!below_turf)
+		our_turf.vis_contents.len = 0
+		if(!add_baseturf_underlay(our_turf))
+			our_turf.ChangeTurf(/turf/open/floor/plating, flags = CHANGETURF_INHERIT_AIR)
+			return FALSE
```

Comment:
```suggestion
			return ELEMENT_INCOMPATIBLE
```
Attach shouldn't have binary return values iirc.

Reply:
This was resolved.